### PR TITLE
[flang] Run real 10 test on x86 only (NFC)

### PR DESCRIPTION
**Pull request by madanial0**
Remove real 10 tests for powerpc

### CODE REVIEW

**Comment by banach-space** [comment]
I don't really contribute to this area, so removing myself from the list of reviewers.

**Comment by kiranchandramohan** [comment]
It is not supported on AArch64 as well.

**Comment by kiranchandramohan** [comment]
> > It is not supported on AArch64 as well.
> 
> Would you like me to add AArch64 macro to that test case as well?

Yes, that would be great. Does it fail currently or give a warning?

File: flang/unittests/Evaluate/real.cpp [hunk 1]
```diff
@@ -14,7 +14,9 @@ using Real2 = Scalar<Type<TypeCategory::Real, 2>>;
 using Real3 = Scalar<Type<TypeCategory::Real, 3>>;
 using Real4 = Scalar<Type<TypeCategory::Real, 4>>;
 using Real8 = Scalar<Type<TypeCategory::Real, 8>>;
+#ifndef __PPC__
```
Comment (luporl) [inline flang/unittests/Evaluate/real.cpp:17]: I think it would be better to use `#ifdef __x86_64__` instead.

**Comment by madanial0** [comment]
Added changes to `flang/test/Lower/assignment.f90`, as it covers the same issue

**Comment by madanial0** [comment]
The macro approach does not seem to be working in the fortran files resulting in a failure will remove the `flang/test/Lower/assignment.f90` test case from this PR

**Comment by clementval** [comment]
> The macro approach does not seem to be working in the fortran files resulting in a failure will remove the `flang/test/Lower/assignment.f90` test case from this PR

Which macro are you using? It might be that the flang driver is not defining them properly.

**Comment by madanial0** [comment]
> > The macro approach does not seem to be working in the fortran files resulting in a failure will remove the `flang/test/Lower/assignment.f90` test case from this PR
> 
> Which macro are you using? It might be that the flang driver is not defining them properly.

I am using `__x86_64__` but it seems not to be picked up by either bbc or flang-new

**Comment by madanial0** [comment]
> > Which macro are you using? It might be that the flang driver is not defining them properly.
> 
> The test is using `bbc` that is a simple test driver that is not defining macros. For this use case, you could try to change `bbc` into `%flang_fc1` (you can replace the -hlfir=false by `-flang-deprecated-no-hlfir`).

I tried the above still did not work with the `__x86_64__` macro, did I miss something?

**Comment by kiranchandramohan** [comment]
> x86_64

The macros are added in the following location.
https://github.com/llvm/llvm-project/blob/dd04e183045f9fb2fb2322df4d623b60c0942850/flang/lib/Frontend/CompilerInvocation.cpp#L1291

**Comment by luporl** [comment]
It seems the `-cpp` flag is needed for this to work.

**Review by kiranchandramohan — APPROVED** [review]
LG.

**Pull request merged**